### PR TITLE
Add enhanceElement Tests

### PR DESCRIPTION
**Pull request by oshalygin**
_This is the initial foray into building out solid test coverage across the library.  Not all functions and components are tested with this commit, this is merely a starting point._

**Add enhanceElement Tests**
* All functions that are consumed internally within the component are now exported so that they can be tested individually with varying test cases to ensure proper functionality.
* The approach to testing taken here is to place test code right next to application code so that it is readily visible which files have test coverage and which dont, regardless of what the coverage report shows.  For this reason, the __test__ folder is deprecated and the associated test files are moved out.
* Instead of using Jest directly, we are leveraging `react-scripts` to run the tests and coverage for us.  This is done intentionally as that library includes the necessary Jest configuration to properly mock dependencies and unmock used dependencies such as lodash.  In the future, that configuration will be pulled into this library and we will deprecate consumption of third party dependencies such as react-scripts.

**Remove test files from build lib**
* The build library should not contain any test files, instead it should contain only source code.  For this reason, test files were removed from the build lib directory

**Update build script**
* The build script is updated to exclude any test files matching the pattern of spec.js.
* Additionally, the chai assertion library is added to provide rich assertions in testing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,6 +77,7 @@
     "babel-preset-es2015": "^6.16.0",
     "babel-preset-react": "^6.16.0",
     "babel-preset-stage-0": "^6.16.0",
+    "chai": "^4.1.0",
```
Comment (tomchentw) [inline package.json]: Please remove `chai` and just leverage `expect` from `jest`.

Comment (oshalygin) [inline package.json]: yeah I'll revert it back, makes sense!

**Comment by oshalygin** [comment]
@tomchentw:

1.  Removed chai
2.  Updated commits to adhere to [standard-version](https://github.com/conventional-changelog/standard-version)

**Comment by tomchentw** [comment]
For merging PR with multiple commits, please:

1. Rebase this branch on the latest `master`
2. In the CLI, checkout to `master` and `git merge THIS_BRANCH`
3. It will be a fast-forward merge and you just `git push` to GitHub, and this PR will be closed.

**Comment by tomchentw** [comment]
Released [v7.3.0](https://github.com/tomchentw/react-google-maps/releases/tag/v7.3.0)

**Pull request merged**